File: src/letcode/LargestGoodInteger.java
package letcode;

public class LargestGoodInteger {
    public static void main(String[] args) {
//        System.out.println(largestGoodInteger("6777133339"));
//        System.out.println(largestGoodInteger("2300019"));
        System.out.println(largestGoodInteger("42352338"));

    }

    public static String largestGoodInteger(String num) {
        StringBuilder ans = new StringBuilder();
        for (int i = 0; i < num.length() - 2; i++) {
            if (num.charAt(i) == num.charAt(i + 1) && num.charAt(i) == num.charAt(i + 2)) {
                if (ans.length() == 0 || ans.charAt(0) < num.charAt(i)) {
                    ans.setLength(0);
                    ans.append(num.charAt(i));
                    ans.append(num.charAt(i));
                    ans.append(num.charAt(i));
                }
                i += 2;
            }
        }
        return ans.toString();
    }
}

File: src/letcode/LargestGoodInteger.kt
package letcode


fun main() {
    print(largestGoodInteger("2300019"))
}

fun largestGoodInteger(num: String): String {
    var ans = ""
    for (i in num.indices) {
        if (i < num.length - 2 && num[i] == num[i + 1] && num[i] == num[i + 2] && (ans.isEmpty() || ans[0] < num[i])) {
            ans = "${num[i]}"
            ans += ans + ans
        }
    }
    return ans
}
